recipes/default.rb

```diff
@@ -59,11 +59,10 @@
 end
 
 kafka_fqdn = consul_helper.get_service_fqdn("kafka")
-broker_port_external = node['kkafka']['broker']['port'].to_i + node['kkafka']['broker']['broker']['id'].to_i
```

Comment:
I don't think this is going to work with multiple nodes as each broker needs to bound on a different port for the external listener

Reply:
If you have ICE@SICS cluster, I think that was weird setup. We've been running on app (or the hackathon cluster) on the same port with no problem

Reply:
@kouzant actually, i ran the tests and it doesn't work for on-premise installation since we register the external listeners with the consul domain name, how does it work for on-premise with external listener using the consul domain?
```
2023-06-23 14:05:23,377 ERROR kafka.Kafka$: Exiting Kafka due to fatal exception during startup.
java.lang.IllegalArgumentException: requirement failed: Configured end points kafka.service.consul:9092 in advertised listeners are already registered by broker 2
	at scala.Predef$.require(Predef.scala:281)
	at kafka.server.KafkaServer.$anonfun$createBrokerInfo$3(KafkaServer.scala:605)
	at kafka.server.KafkaServer.$anonfun$createBrokerInfo$3$adapted(KafkaServer.scala:603)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at kafka.server.KafkaServer.createBrokerInfo(KafkaServer.scala:603)
	at kafka.server.KafkaServer.startup(KafkaServer.scala:354)
	at kafka.Kafka$.main(Kafka.scala:115)
	at kafka.Kafka.main(Kafka.scala)
```

Reply:
Good catch, it makes sense if the host is the same to complain. In that case, I think it has to be incremented. For the Cloud they have to listen on the same port, otherwise we will have to open more ports on the security group/firewall.